mreg_cli/network.py

```diff
@@ -203,6 +185,168 @@ def info(args):
+        "category",
+        "location",
+        "frozen",
+        "reserved",
+    ]
+    for name in param_names:
+        value = args_dict.get(name)
+        if value is None:
+            continue
+        param, val = convert_wildcard_to_regex(name, value)
+        params[param] = val
+
+    if not params:
+        cli_warning("Need at least one search criteria")
+
+    path = f"/api/v1/networks/"
+    networks = get_list(path, params)
+    if not networks:
+        cli_warning("No networks matching the query were found.")
+
+    n_networks = len(networks)
+    for i, nwork in enumerate(networks):
+        if args.limit and i >= args.limit:
+            omitted = n_networks - i
+            s = "s" if omitted > 1 else ""
+            if not args.silent:
+                print(f"Reached limit ({args.limit}). Omitted {omitted} network{s}.")
+            break
+        if args.addr_only:
+            print(nwork["network"])
+        else:
+            print_network_info(nwork)
+            print()  # Blank line between networks
+
+    s = "s" if n_networks > 1 else ""
+    if not args.silent:
+        print(f"Found {n_networks} network{s} matching the search criteria.")
+
+
+network.add_command(
+    prog="find",
+    description="Search for networks based on a range of search parameters.",
+    short_desc="Search for networks.",
+    callback=find,
+    flags=[
+        Flag(
+            "-network",
+            description="Network address.",
+            metavar="NETWORK",
+        ),
+        Flag(
+            "-description",
+            description="Description.",
+            metavar="DESCRIPTION",
+        ),
+        Flag(
+            "-vlan",
+            description="VLAN.",
+            metavar="VLAN",
+        ),
+        Flag(
+            "-dns_delegated",
+            description="DNS delegation status (0 or 1).",
+            metavar="DNS-DELEGATED",
+        ),
+        Flag(
+            "-category",
+            description="Category.",
+            metavar="CATEGORY",
+        ),
+        Flag(
+            "-location",
+            description="Location.",
+            metavar="LOCATION",
+        ),
+        Flag(
+            "-frozen",
+            description="Frozen status (0 or 1).",
+            metavar="FROZEN",
+        ),
+        Flag(
+            "-reserved",
+            description="Exact number of reserved network addresses.",
+            metavar="RESERVED",
+        ),
+        Flag(
+            "-addr-only",
+            description="Only print network address of matching networks.",
+            action="store_true",
+        ),
+        Flag(
+            "-limit",
+            description="Maximum number of networks to print.",
+            metavar="LIMIT",
+            type=int,
+        ),
+        Flag(
+            "-silent",
+            description="Do not print meta info (number of networks found, limit reached, etc.).",
+            action="store_true",
+        ),
+    ],
+)
+
```

Comment:
`network info` also can find a network given an ip in its range, to match you example `network info 129.199.100.9` .
Maybe an option `-ip` to search for a network with a given IP?
`get_network_range_from_input` not only handles CIDRs or network addresses, but any ip address.